### PR TITLE
Metering API Implementation

### PR DESCRIPTION
**Pull request by stu-elastic**
This pulls in @pgomulka's branch [stu-elastic-otel-meter-api](https://github.com/pgomulka/elasticsearch/tree/stu-elastic-otel-meter-api) and adds the elasticsearch versions of the otel instruments.

### CODE REVIEW

File: distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java [hunk 1]
```diff
@@ -50,10 +50,12 @@ class APMJvmOptions {
         // by the agent. Don't disable writing to a log file, as the agent will then
         // require extra Security Manager permissions when it tries to do something
         // else, and it's just painful.
-        "log_file", "_AGENT_HOME_/../../logs/apm.log",
-
+        "log_file", "/Users/przemyslawgomulka/workspace/pgomulka/apm.log",
```
Comment (rjernst) [inline distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java:53]: The log file and level need to be reset back to their original levels.

**Review by rjernst — COMMENTED** [review]
I'm not sure which is the correct PR to look at right now, but I had these comments from yesterday that I thought I would post. I only got through a small portion of the PR; this is not a complete review.

File: modules/apm/build.gradle [hunk 2]
```diff
@@ -14,11 +14,17 @@ esplugin {
...
+    name "sonatype-nexus-snapshots"
+    url "https://oss.sonatype.org/content/repositories/snapshots"
+  }
+}
 dependencies {
   implementation "io.opentelemetry:opentelemetry-api:${otelVersion}"
   implementation "io.opentelemetry:opentelemetry-context:${otelVersion}"
   implementation "io.opentelemetry:opentelemetry-semconv:${otelVersion}-alpha"
-  runtimeOnly    "co.elastic.apm:elastic-apm-agent:1.36.0"
+  implementation    "co.elastic.apm:elastic-apm-agent:1.42.1-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (stu-elastic) [reply modules/apm/build.gradle]: Sounds good.

**Comment by stu-elastic** [comment]
Superseded by https://github.com/elastic/elasticsearch/pull/99832

**Pull request closed without merging**